### PR TITLE
ORC: Add configurable write properties

### PR DESCRIPTION
**Pull request by hililiwei**
closes https://github.com/apache/iceberg/issues/3272

> To be consistent with Parquet, it would be good to introduce write table properties for ORC too.
Also make sure that we have a configuration available for ORC separately.
Discussed here: #2935 (comment)

### CODE REVIEW

**Comment by hililiwei** [comment]
reopen rerun CI

File: core/src/main/java/org/apache/iceberg/TableProperties.java [hunk 1]
```diff
@@ -111,6 +111,12 @@ private TableProperties() {
   public static final String DELETE_AVRO_COMPRESSION = "write.delete.avro.compression-codec";
   public static final String AVRO_COMPRESSION_DEFAULT = "gzip";
 
+  public static final String ORC_STRIPE_SIZE_BYTES = "write.orc.stripe-size-bytes";
+  public static final String ORC_STRIPE_SIZE_BYTES_DEFAULT = "67108864"; // 64L * 1024 * 1024
```
Comment (jackye1995) [inline core/src/main/java/org/apache/iceberg/TableProperties.java:115]: nit: can directly be `long`

File: core/src/main/java/org/apache/iceberg/TableProperties.java [hunk 2]
```diff
@@ -111,6 +111,12 @@ private TableProperties() {
...
+
+  public static final String ORC_BLOCK_SIZE_BYTES = "write.orc.block-size-bytes";
+  public static final String ORC_BLOCK_SIZE_BYTES_DEFAULT = "268435456"; // 256L * 1024 * 1024
```
Comment (jackye1995) [inline core/src/main/java/org/apache/iceberg/TableProperties.java:118]: nit: can directly be `long`

File: site/docs/configuration.md [hunk 3]
```diff
@@ -41,6 +41,8 @@ Iceberg tables support table properties to configure table behavior, like the de
 | write.parquet.compression-codec    | gzip               | Parquet compression codec                          |
 | write.parquet.compression-level    | null               | Parquet compression level                          |
 | write.avro.compression-codec       | gzip               | Avro compression codec                             |
+| write.orc.stripe-size-bytes | 67108864(64 MB) | Define the default ORC stripe size, in bytes. |
```
Comment (jackye1995) [inline site/docs/configuration.md:44]: nit: align `|` for each column unless it's too long.

File: orc/src/main/java/org/apache/iceberg/orc/ORC.java [hunk 4]
```diff
@@ -145,6 +146,7 @@ public WriteBuilder metricsConfig(MetricsConfig newMetricsConfig) {
 
     public <D> FileAppender<D> build() {
       Preconditions.checkNotNull(schema, "Schema is required");
+      parseWriterCustomizedConf(conf);
```
Comment (jackye1995) [inline orc/src/main/java/org/apache/iceberg/orc/ORC.java:149]: it seems not necessary to have a separated method for this

File: orc/src/main/java/org/apache/iceberg/orc/ORC.java [hunk 5]
```diff
@@ -511,4 +513,14 @@ static Reader newFileReader(InputFile file, Configuration config) {
...
   }
+
+  private static void parseWriterCustomizedConf(Configuration conf) {
+    conf.set(
+        OrcConf.STRIPE_SIZE.name(),
+        conf.get(TableProperties.ORC_STRIPE_SIZE_BYTES, TableProperties.ORC_STRIPE_SIZE_BYTES_DEFAULT));
+
+    conf.set(OrcConf.BLOCK_SIZE.name(), conf.get(
+        TableProperties.ORC_BLOCK_SIZE_BYTES,
+        TableProperties.ORC_BLOCK_SIZE_BYTES_DEFAULT));
```
Comment (jackye1995) [inline orc/src/main/java/org/apache/iceberg/orc/ORC.java:524]:
```suggestion
    conf.set(
        OrcConf.BLOCK_SIZE.name(), 
        conf.get(TableProperties.ORC_BLOCK_SIZE_BYTES, TableProperties.ORC_BLOCK_SIZE_BYTES_DEFAULT));
```

File: core/src/main/java/org/apache/iceberg/TableProperties.java (hunk 1 above)
Comment (hililiwei) [reply core/src/main/java/org/apache/iceberg/TableProperties.java]: done

File: core/src/main/java/org/apache/iceberg/TableProperties.java (hunk 2 above)
Comment (hililiwei) [reply core/src/main/java/org/apache/iceberg/TableProperties.java]: done

File: site/docs/configuration.md (hunk 3 above)
Comment (hililiwei) [reply site/docs/configuration.md]: done

File: orc/src/main/java/org/apache/iceberg/orc/ORC.java (hunk 4 above)
Comment (hililiwei) [reply orc/src/main/java/org/apache/iceberg/orc/ORC.java]: removed

File: orc/src/main/java/org/apache/iceberg/orc/ORC.java (hunk 5 above)
Comment (hililiwei) [reply orc/src/main/java/org/apache/iceberg/orc/ORC.java]: done

File: orc/src/main/java/org/apache/iceberg/orc/ORC.java (hunk 4 above)
Comment (hililiwei) [reply orc/src/main/java/org/apache/iceberg/orc/ORC.java]: @jackye1995  Would it be more reasonable to add a common method to resolve all configuration items, converting the specific name in Iceberg to the native name of orc? In this way, users can use all orc configuration items, not only those listed in properties.
The method may like this:

```
  private void setAllOrcConf(Configuration conf) {
    Arrays.stream(OrcConf.values()).forEach(oneConf -> {
      String attribute = oneConf.getAttribute();
      String icebergPropertiesName =
          ("write." + attribute).replace(".", "-")
              .replaceFirst("-", ".")
              .replaceFirst("-", ".");
      if (conf.get(icebergPropertiesName) != null) {
        conf.set(attribute, conf.get(icebergPropertiesName);
      }
    });
  }
```

File: core/src/main/java/org/apache/iceberg/TableProperties.java [hunk 6]
```diff
@@ -111,6 +111,12 @@ private TableProperties() {
   public static final String DELETE_AVRO_COMPRESSION = "write.delete.avro.compression-codec";
   public static final String AVRO_COMPRESSION_DEFAULT = "gzip";
 
+  public static final String ORC_STRIPE_SIZE_BYTES = "write.orc.stripe-size-bytes";
+  public static final long ORC_STRIPE_SIZE_BYTES_DEFAULT = 64L * 1024 * 1024; // 64 MB
+
+  public static final String ORC_BLOCK_SIZE_BYTES = "write.orc.block-size-bytes";
```
Comment (rdblue) [inline core/src/main/java/org/apache/iceberg/TableProperties.java:117]: @omalley, is this a config that we should expose as an Iceberg table setting? Is it still needed for object stores?

**Comment by pvary** [comment]
I think if we decide  that we need the default configuration values set for Iceberg reads, then we should do this the same way as it has been done with the other file formats:
- Parquet:
https://github.com/apache/iceberg/blob/129cdfafdedb3df157fe925023ce4741dba93986/parquet/src/main/java/org/apache/iceberg/parquet/Parquet.java#L304-L320
- Avro:
https://github.com/apache/iceberg/blob/303f92547c344146a3ea500e315c285fa446d7b2/core/src/main/java/org/apache/iceberg/avro/Avro.java#L208-L213

Basically we should create a Context object for the ORC as well.

**Comment by hililiwei** [comment]
> I think if we decide that we need the default configuration values set for Iceberg reads, then we should do this the same way as it has been done with the other file formats:
> …………
> Basically we should create a Context object for the ORC as well.

added to the latest commiter.  PTAL, thx.

**Comment by pvary** [comment]
@hililiwei: Do we have any tests for this in Parquet/Avro?
I think it would b good to add some tests, so we can be sure that these configurations are working as expected.

**Comment by hililiwei** [comment]
> @hililiwei: Do we have any tests for this in Parquet/Avro? I think it would b good to add some tests, so we can be sure that these configurations are working as expected.

@pvary Checked, and seemed  no relevant tests. I added one unit test for ORC this time. How about let me open another PR for Parquet\AVRO?

**Comment by pvary** [comment]
@szlta: Would you mind taking a look at the tests?

Thanks,
Peter

File: flink/v1.14/flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSink.java [hunk 7]
```diff
@@ -58,11 +58,11 @@
   @ClassRule
   public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
 
-  private static final String TABLE_NAME = "test_table";
+  public static final String TABLE_NAME = "test_table";
```
Comment (rdblue) [inline flink/v1.14/flink/src/test/java/org/apache/iceberg/flink/TestFlinkTableSink.java:61]: Why are these changes necessary?

File: flink/v1.14/flink/src/test/java/org/apache/iceberg/flink/TestOrcTableProperties.java [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+  public TestOrcTableProperties(
+      String catalogName,
+      Namespace baseNamespace,
+      FileFormat format,
+      Boolean isStreamingJob) {
+    super(catalogName, baseNamespace, format, isStreamingJob);
+  }
+
+  @Parameterized.Parameters(name = "catalogName={0}, baseNamespace={1}, format={2}, isStreaming={3}")
+  public static Iterable<Object[]> parameters() {
```
Comment (rdblue) [inline flink/v1.14/flink/src/test/java/org/apache/iceberg/flink/TestOrcTableProperties.java:53]: I doubt that it is worth the cost of this test. As for Parquet and Avro, the integration test is expensive and not very useful. Instead, this should test that the properties get translated and passed into ORC correctly.

File: orc/src/main/java/org/apache/iceberg/orc/ORC.java [hunk 9]
```diff
@@ -99,8 +101,9 @@ public WriteBuilder metadata(String property, String value) {
 
     /**
      * Setting a specific configuration value for the writer.
+     *
      * @param property The property to set
-     * @param value The value to set
+     * @param value    The value to set
```
Comment (rdblue) [inline orc/src/main/java/org/apache/iceberg/orc/ORC.java:106]: Please revert unnecessary whitespace changes.

File: flink/v1.14/flink/src/test/java/org/apache/iceberg/flink/TestOrcTableProperties.java (hunk 8 above)
Comment (hililiwei) [reply flink/v1.14/flink/src/test/java/org/apache/iceberg/flink/TestOrcTableProperties.java]: modified like https://github.com/apache/iceberg/pull/3902

File: docs/versioned/tables/configuration.md [hunk 10]
```diff
@@ -52,6 +52,8 @@ Iceberg tables support table properties to configure table behavior, like the de
 | write.parquet.compression-level    | null               | Parquet compression level                          |
 | write.avro.compression-codec       | gzip               | Avro compression codec: gzip(deflate with 9 level), gzip, snappy, uncompressed |
 | write.avro.compression-level       | null               | Avro compression level                             |
+| write.orc.stripe-size-bytes        | 67108864 (64 MB)   | Define the default ORC stripe size, in bytes       |
```
Comment (openinx) [inline docs/versioned/tables/configuration.md:55]: Looks like the document don't add any delete write configure keys, right ?  I think we should publish another separate PR to add those delete related properties.

File: orc/src/main/java/org/apache/iceberg/orc/ORC.java [hunk 11]
```diff
@@ -143,12 +152,73 @@ public WriteBuilder metricsConfig(MetricsConfig newMetricsConfig) {
...
+      private Context(long stripeSize, long blockSize) {
+        this.stripeSize = stripeSize;
+        this.blockSize = blockSize;
+      }
+
+      static Context dataContext(Configuration conf) {
+        String stripeSizeStr =
+            conf.get(TableProperties.ORC_STRIPE_SIZE_BYTES, conf.get(OrcConf.STRIPE_SIZE.getAttribute()));
+        long stripeSize = stripeSizeStr != null ? Long.parseLong(stripeSizeStr) :
+            TableProperties.ORC_STRIPE_SIZE_BYTES_DEFAULT;
```
Comment (openinx) [inline orc/src/main/java/org/apache/iceberg/orc/ORC.java:198]: Why not just use `conf.getLong(key,  defaultValue)` ?

File: orc/src/main/java/org/apache/iceberg/orc/ORC.java [hunk 12]
```diff
@@ -143,12 +152,73 @@ public WriteBuilder metricsConfig(MetricsConfig newMetricsConfig) {
...
+
+        String blockSizeStr = conf.get(
+            TableProperties.ORC_BLOCK_SIZE_BYTES,
+            conf.get(OrcConf.BLOCK_SIZE.getAttribute()));
+
+        long blockSize = blockSizeStr != null ? Long.parseLong(blockSizeStr) :
+            TableProperties.ORC_BLOCK_SIZE_BYTES_DEFAULT;
```
Comment (openinx) [inline orc/src/main/java/org/apache/iceberg/orc/ORC.java:205]: Ditto.

File: orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java [hunk 13]
```diff
@@ -0,0 +1,81 @@
...
+  public static final Schema SCHEMA = new Schema(
+      Types.NestedField.optional(1, "id", Types.IntegerType.get()),
+      Types.NestedField.optional(2, "data", Types.StringType.get())
+  );
+
+  @ClassRule
+  public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
+
+  @Test
+  public void testAvroProperties() throws Exception {
```
Comment (openinx) [inline orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java:53]: `testAvroProperties` ?  It's an ORC unit test,  right ?

File: orc/src/main/java/org/apache/iceberg/orc/ORC.java [hunk 14]
```diff
@@ -143,12 +152,73 @@ public WriteBuilder metricsConfig(MetricsConfig newMetricsConfig) {
...
+      }
+
+      private Context(long stripeSize, long blockSize) {
+        this.stripeSize = stripeSize;
+        this.blockSize = blockSize;
+      }
+
+      static Context dataContext(Configuration conf) {
+        String stripeSizeStr =
+            conf.get(TableProperties.ORC_STRIPE_SIZE_BYTES, conf.get(OrcConf.STRIPE_SIZE.getAttribute()));
```
Comment (openinx) [inline orc/src/main/java/org/apache/iceberg/orc/ORC.java:196]: What's the difference between the `getAttribute` and `getHiveConfName` ?  Should fallback to read the hive conf key if the attribute is not present in this hadoop configuraiton ?

File: orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java [hunk 15]
```diff
@@ -0,0 +1,81 @@
...
+
+    ORC.WriteBuilder writeBuilder = spy(ORC.write(Files.localOutput(TEMPORARY_FOLDER.newFile())));
+    writeBuilder.forTable(table);
+    ArgumentCaptor<Map<String, String>> argument = ArgumentCaptor.forClass(Map.class);
+    verify(writeBuilder).setAll(argument.capture());
+    Map<String, String> config = argument.getValue();
+
+    Assert.assertEquals(String.valueOf(stripeSizeBytes), config.get(TableProperties.ORC_STRIPE_SIZE_BYTES));
+    Assert.assertEquals(String.valueOf(blockSizeBytes), config.get(TableProperties.ORC_BLOCK_SIZE_BYTES));
+    Assert.assertEquals(FileFormat.ORC.name(), config.get(TableProperties.DEFAULT_FILE_FORMAT));
```
Comment (openinx) [inline orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java:79]: I don't think check the `Map<String, String>` table properties is enough for this PR, because it actually don't cover the the `WriteBuilder.Context::deleteContext` or `WriteBuilder.Context::dataContext` .  the correct approach should be checking the values of `OrcConf.STRIPE_SIZE` & `OrcConf.BLOCK_SIZE` from hadoop configuration.

File: orc/src/main/java/org/apache/iceberg/orc/ORC.java (hunk 11 above)
Comment (hililiwei) [reply orc/src/main/java/org/apache/iceberg/orc/ORC.java:198]: If the value of `TableProperties.ORC_STRIPE_SIZE_BYTES` is null, first attempts to obtain the value of native `OrcConf.STRIPE_SIZE`. `TableProperties.ORC_STRIPE_SIZE_BYTES_DEFAULT` is used only when neither `TableProperties.ORC_STRIPE_SIZE_BYTES` nor `OrcConf.STRIPE_SIZE` is configured.

Comment (hililiwei) [reply orc/src/main/java/org/apache/iceberg/orc/ORC.java:198]: This logic is used to prevent users from using native configuration instead of iceberg config. I refer to the implementation of other configuration items, which is more reasonable to remove.
Removed.

File: orc/src/main/java/org/apache/iceberg/orc/ORC.java (hunk 14 above)
Comment (hililiwei) [reply orc/src/main/java/org/apache/iceberg/orc/ORC.java]: Whether the following logic should be used:
1. First try to get `TableProperties.ORC_STRIPE_SIZE_BYTES`
2. If null, try `OrcConf.STRIPE_SIZE.getAttribute()`.
3. If `getAttribute` is null, try `getHiveConfName`.

File: orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+  public static final TemporaryFolder TEMPORARY_FOLDER = new TemporaryFolder();
+
+  @Test
+  public void testAvroProperties() throws Exception {
+    Long stripeSizeBytes = 32L * 1024 * 1024;
+    Long blockSizeBytes = 128L * 1024 * 1024;
+
+    ImmutableMap<String, String> properties = ImmutableMap.of(
+        TableProperties.ORC_STRIPE_SIZE_BYTES, String.valueOf(stripeSizeBytes),
+        TableProperties.ORC_BLOCK_SIZE_BYTES, String.valueOf(blockSizeBytes),
```
[the remark this answers is not in the document]
Comment (hililiwei) [reply orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java:61]: done.

File: orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java (hunk 15 above)
Comment (hililiwei) [reply orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java]: Reasonable. Modified. Thx.

File: orc/src/main/java/org/apache/iceberg/orc/ORC.java (hunk 14 above)
Comment (openinx) [reply orc/src/main/java/org/apache/iceberg/orc/ORC.java:196]: I mean if people just set their hiveConfName but don't set the key attribution name, then should we also parse the hiveConfName , right ?

Comment (openinx) [reply orc/src/main/java/org/apache/iceberg/orc/ORC.java:196]: I think the correct approach to parse the config keys is: 

```java
static Context dataContext(Configuration conf) {
        long stripeSize = TableProperties.ORC_STRIPE_SIZE_BYTES_DEFAULT;
        stripeSize = conf.getLong(OrcConf.STRIPE_SIZE.getAttribute(), stripeSize);
        stripeSize = conf.getLong(OrcConf.STRIPE_SIZE.getHiveConfName(), stripeSize);
        stripeSize = conf.getLong(TableProperties.ORC_STRIPE_SIZE_BYTES, stripeSize);

        long blockSize = TableProperties.ORC_BLOCK_SIZE_BYTES_DEFAULT;
        blockSize = conf.getLong(OrcConf.BLOCK_SIZE.getAttribute(), blockSize);
        blockSize = conf.getLong(OrcConf.BLOCK_SIZE.getHiveConfName(), blockSize);
        blockSize = conf.getLong(TableProperties.ORC_BLOCK_SIZE_BYTES, blockSize);

        return new Context(stripeSize, blockSize);
      }
```

File: orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java [hunk 17]
```diff
@@ -0,0 +1,131 @@
...
+
+    File testFile = TEMPORARY_FOLDER.newFile();
+    Assert.assertTrue(testFile.delete());
+
+    EqualityDeleteWriter<Object> deleteWriter = ORC.writeDeletes(Files.localOutput(testFile))
+        .forTable(table)
+        .equalityFieldIds(1)
+        .createWriterFunc(GenericOrcWriter::buildWriter)
+        .buildEqualityWriter();
+
```
Comment (openinx) [inline orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java:117]: Nit: Please remove this extra empty line here.

File: orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java [hunk 18]
```diff
@@ -0,0 +1,131 @@
...
+
+    DynFields.BoundField<OrcFileAppender> writer =
+        DynFields.builder().hiddenImpl(deleteWriter.getClass(), "appender").build(deleteWriter);
+
+    OrcFileAppender orcFileAppender = writer.get();
+    DynFields.BoundField<Configuration> confField =
+        DynFields.builder().hiddenImpl(orcFileAppender.getClass(), "conf").build(orcFileAppender);
```
Comment (openinx) [inline orc/src/test/java/org/apache/iceberg/orc/TestTableProperties.java:124]: Nit: Please use the `OrcFileAppender<Record>` here to eliminate the idea warnings.

**Review by openinx — COMMENTED** [review]
Almost looks good to me overall,  just left several comments.

File: orc/src/main/java/org/apache/iceberg/orc/ORC.java (hunk 14 above)
Comment (hililiwei) [reply orc/src/main/java/org/apache/iceberg/orc/ORC.java:196]: Got it. it is much more graceful.

File: docs/versioned/tables/configuration.md (hunk 10 above)
Comment (hililiwei) [reply docs/versioned/tables/configuration.md:55]: I will raise a PR for this soon.

**Review by openinx — APPROVED** [review]
Looks good to me now, thanks @hililiwei for the contribution, and thanks @rdblue and @jackye1995 for the reviewing !

**Comment by hililiwei** [comment]
Thank you all for reviewing.

**Comment by rdblue** [comment]
@openinx, @hililiwei, I think we need to revert this. It looks like the `dataContext` and `deleteContext` methods accept Hadoop Configuration objects rather than property maps.

Iceberg doesn't use Hadoop Configuration. We can either directly fix this to work like the Parquet contexts with another PR, or I think we should revert and open a replacement PR. What do you think is the right way forward?

**Comment by hililiwei** [comment]
> @openinx, @hililiwei, I think we need to revert this. It looks like the `dataContext` and `deleteContext` methods accept Hadoop Configuration objects rather than property maps.
> 
> Iceberg doesn't use Hadoop Configuration. We can either directly fix this to work like the Parquet contexts with another PR, or I think we should revert and open a replacement PR. What do you think is the right way forward?

I understand what you mean roughly. Let me mention a separate PR, it may goes beyond the this PR content.

**Comment by openinx** [comment]
@rdblue  The `ORC#WriterBuilder` will copy all the key values from property map to the Hadoop configuration, please see [1] and [2]. So in my thought,  we can just use the unified hadoop configuration to get all of the configure keys.  Will this answer your concern ? 

[1]. https://github.com/apache/iceberg/blob/master/orc/src/main/java/org/apache/iceberg/orc/ORC.java#L122
[2]. https://github.com/apache/iceberg/blob/master/orc/src/main/java/org/apache/iceberg/orc/ORC.java#L132

**Comment by openinx** [comment]
Let's make this more clear: 

In the ORC class, we just copied all key values from property maps to hadoop configuration. And all the following configure keys will be parsed from hadoop configuration instance. 

In the current Parquet class, we will just keep all the property maps into in-memory hash map, and then the `dataContext` & `deleteContext` will parse the hash map directly.  Finally,  we will copy the key values from HashMap to hadoop configuration.

In my mind, I just don't see the difference between the two approach. Both look good to me.

**Comment by zhongyujiang** [comment]
@openinx, I think what @rdblue means is that Iceberg configuration should not be overridden by Hadoop configuration, for example:
when Hadoop configuraion is configed a property 
"orc.compress" = "snappy"
but no relevant property has been configed in Iceberg table props, In the present context ORC will use snappy for compress as configed in Hadoop. But the correct way is use default codec configed in `TableProperties` when no relevant property has been set.

So we need a props map to receive table props passed in, then set relevant ORC props in Hadoop configuration after parsing, just like Parquet and Avro.

**Comment by rdblue** [comment]
For Parquet the Hadoop configuration is used to pass options into the data file. It is not used as the source of table configuration. Table configuration properties should never come from the Hadoop Configuration.

The steps should be:
1. Get the Hadoop configuration, if it is present in the InputFile or OutputFile. If not, default it
2. Get config from the builder and the table properties
3. Set configuration from step 2 on the Hadoop config
4. Create the reader or writer with the Hadoop config

The only configuration coming from the Hadoop Configuration itself is whatever was in the environment.

One possibly confusing thing is that we also set config values directly in the Hadoop configuration. That handles cases where the user wants to pass config properties that are not standardized in Iceberg. So you could use `set("parquet.bloom.filter.enabled#id", "true")` for example. Standardized table settings should override these.

**Comment by openinx** [comment]
Okay,  I think I get the drawback about the current ORC builder ( which copied the table properties map into hadoop conf and just use the hadoop conf as the source of truth): 

* As @zhongyujiang said, the default value of iceberg standardized settings will be replaced by the hadoop config values.
* Non-standardized iceberg settings which are added in the iceberg table properties will be affected to the underlying ORC readers & writers. 
* The standardized iceberg settings will be added to hadoop configuration, which may pollute the hadoop configuration.

I agree to make another PR to change the ORC configs parser as the @rdblue suggested.

Thanks.
